### PR TITLE
[Doc] refactor flink docs on docs web to be consistent with flink docs in code repo

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/loading/Flink-connector-starrocks.md [hunk 1]
```diff
@@ -1,173 +1,647 @@
...
-<InsertPrivNote />
+| Connector | Flink                    | StarRocks     | Java | Scala     |
+|-----------|--------------------------|---------------| ---- |-----------|
+| 1.2.7     | 1.11,1.12,1.13,1.14,1.15 | 2.1 and later | 8    | 2.11,2.12 |
 
-## Overview
+## Obtain Flink connector
 
-The flink-connector-jdbc tool provided by Apache Flink® may not meet your performance requirements in certain scenarios. Therefore we provide a new connector named flink-connector-starrocks, which can cache data and then load data at a time by using [Stream Load](./StreamLoad.md).
+You can obtain the Flink connector JAR file in the following ways:
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:14]: image
"Flink connector" or "Flink Connector"? Adhere to one usage throughout the doc.

File: docs/loading/Flink-connector-starrocks.md [hunk 2]
```diff
@@ -1,173 +1,647 @@
...
 
-## Procedure
+- Directly download the compiled Flink Connector JAR file.
+- Add the Flink connector as a dependency in your Maven project and then download the JAR file.
+- Compile the source code of the Flink Connector into a JAR file by yourself.
 
-To load data from Apache Flink® into StarRocks by using flink-connector-starrocks, perform the following steps:
+The naming format of the Flink connector JAR file
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:20]: image
Add a colon at the end of the sentence and use the bold format. Or change the sentence to "The naming format of the Flink connector JAR file is as follows:"

File: docs/loading/Flink-connector-starrocks.md [hunk 3]
```diff
@@ -1,173 +1,647 @@
...
 
-1. Download the [source code](https://github.com/StarRocks/flink-connector-starrocks) of flink-connector-starrocks.
-2. Find a file named **pom.xml**. Add the following code snippet to **pom.xml** and replace `x.x.x` in the code snippet with the latest version number of flink-connector-starrocks.
+- Since Flink 1.15, it's `flink-connector-starrocks-${connector_version}_flink-${flink_version}.jar`. For example, if you install Flink 1.15 and you want to use Flink connector 1.2.7, you can use `flink-connector-starrocks-1.2.7_flink-1.15.jar`
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:22]: image
The period (.) is missing

File: docs/loading/Flink-connector-starrocks.md [hunk 4]
```diff
@@ -1,173 +1,647 @@
...
-    ```Plain%20Text
+- Prior to Flink 1.15, it's `flink-connector-starrocks-${connector_version}_flink-${flink_version}_${scala_version}.jar`. For example, if you install Flink 1.14 and Scala 2.12 in your environment, and you want to use Flink connector 1.2.7, you can use `flink-connector-starrocks-1.2.7_flink-1.14_2.12.jar`.
+
+> **NOTICE**
+>
+> In general, the latest version of the Flink connector only maintains compatibility with the three most recent versions of Flink.
+
+### Download the compiled Jar file
+
+Directly download the corresponding version of the Flink connector JAR from the [Maven Central Repository](https://repo1.maven.org/maven2/com/starrocks).
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:32]: image
Add "file" following "JAR". It should be "JAR file".

File: docs/loading/Flink-connector-starrocks.md [hunk 5]
```diff
@@ -1,173 +1,647 @@
...
+
+### Maven Dependency
+
+In your Maven project's `pom.xml` file, add the Flink connector as a dependency according to the following format. Replace `flink_version`, `scala_version`, and `connector_version` with the respective versions.
+
+- Since Flink 1.15
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:38]: -> In Flink 1.15 and later

File: docs/loading/Flink-connector-starrocks.md [hunk 6]
```diff
@@ -1,173 +1,647 @@
...
-| sink.buffer-flush.max-bytes   | No           | 94371840(90M)     | STRING        | The maximum size of data that can be loaded into StarRocks at a time. Valid values: 64 MB to 10 GB. |
-| sink.buffer-flush.max-rows    | No           | 500000            | STRING        | The maximum number of rows that can be loaded into StarRocks at a time. Valid values: 64000 to 5000000. |
-| sink.buffer-flush.interval-ms | No           | 300000            | STRING        | The interval at which data is flushed. Valid values: 1000 to 3600000. Unit: ms. |
-| sink.max-retries              | No           | 3                 | STRING        | The number of times that the system retries to perform the Stream Load. Valid values: 0 to 10. |
-| sink.connect.timeout-ms       | No           | 1000              | STRING        | The period of time after which the stream load times out. Valid values: 100 to 60000. Unit: ms. |
-| sink.properties.*             | No           | NONE              | STRING        | The properties of the stream load. The properties include k1, k2, and k3. Since 2.4, the flink-connector-starrocks supports partial updates for Primary Key tables. |
+- Prior to Flink 1.15
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:48]: -> In versions earlier than Flink 1.15

File: docs/loading/Flink-connector-starrocks.md [hunk 7]
```diff
@@ -1,173 +1,647 @@
...
+        <groupId>com.starrocks</groupId>
+        <artifactId>flink-connector-starrocks</artifactId>
+        <version>${connector_version}_flink-${flink_version}_${scala_version}</version>
+    </dependency>
+    ```
+
+### Compile by yourself
+
+1. Download the [Flink connector package](https://github.com/StarRocks/starrocks-connector-for-apache-flink).
+2. Execute the following command to compile the source code of Flink connector into a JAR file. Note that  `flink_version` is replaced with the corresponding Flink version.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:61]: image
extra space

File: docs/loading/Flink-connector-starrocks.md [hunk 8]
```diff
@@ -1,173 +1,647 @@
...
+| sink.version                      | No           | AUTO              | Supported since 1.2.4. Choose which implementation to use. <ul><li>`V1`: use [Stream Load](https://docs.starrocks.io/en-us/latest/loading/StreamLoad) interface to load data. Connectors before 1.2.4 only support this mode. </li> <li>`V2`: use [Transaction Stream Load](https://docs.starrocks.io/en-us/latest/loading/Stream_Load_transaction_interface) interface to load data. It requires StarRocks to be at least version 2.4. Recommends `V2` because it optimizes the memory usage and provide a more stable exactly-once implementation. </li> <li>`AUTO`: if the version of StarRocks supports transaction stream load, will choose `V2` automatically, otherwise choose `V1` </li></ul> |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:90]: image
Give the param description first, and then tell the user from which version this param is supported. -> The implementation to use. This parameter is supported from 1.2.4 onwards.

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:90]: image
-> provides

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:90]: image
Capitalize the first letter of these words

File: docs/loading/Flink-connector-starrocks.md [hunk 9]
```diff
@@ -1,173 +1,647 @@
...
+| sink.buffer-flush.max-bytes       | No           | 94371840(90M)     | The maximum size of data that can be accumulated in memory before being sent to StarRocks at a time. Valid values: 64 MB to 10 GB. Setting this parameter to a larger value can improve loading performance but may increase loading latency.                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:93]: image
These are not "values", none of them are. Suggest changing the sentence to -> The maximum value ranges from 64 MB to 10 GB.

File: docs/loading/Flink-connector-starrocks.md [hunk 10]
```diff
@@ -1,173 +1,647 @@
...
+| sink.buffer-flush.max-rows        | No           | 500000            | The maximum number of rows that can be accumulated in memory before being sent to StarRocks at a time. Only available for version `V1` when using at-least-once. Valid values: 64000 to 5000000.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:94]: image
-> This parameter is available only when you set `sink.version` to `V1` and set `sink.semantic` to `at-least-once`.

File: docs/loading/Flink-connector-starrocks.md [hunk 11]
```diff
@@ -1,173 +1,647 @@
...
+| sink.buffer-flush.interval-ms     | No           | 300000            | The interval at which data is flushed. Only available for at-least-once mode. Valid values: 1000 to 3600000. Unit: ms.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:95]: image
See the comment on Line 94.

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:95]: image
Wrap these values in code format. There are many others, make sure you see to all of them in the doc.

File: docs/loading/Flink-connector-starrocks.md [hunk 12]
```diff
@@ -1,173 +1,647 @@
...
+| sink.max-retries                  | No           | 3                 | The number of times that the system retries to perform the Stream Load. Only available for sink version `V1` currently. Valid values: 0 to 10.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:96]: image
I guess it should be "the Stream Load job"?

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:96]: image
See Line 94.

File: docs/loading/Flink-connector-starrocks.md [hunk 13]
```diff
@@ -1,173 +1,647 @@
...
+| sink.wait-for-continue.timeout-ms | No           | 10000             | Supported since 1.2.7. The timeout for waiting response of HTTP 100-continue from FE. Valid values: 3000 to 600000. Unit: ms                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:98]: image
-> the FE

File: docs/loading/Flink-connector-starrocks.md [hunk 14]
```diff
@@ -1,173 +1,647 @@
...
+| sink.ignore.update-before         | No           | true              | Supported since version 1.2.8. Whether to ignore `UPDATE_BEFORE` records from Flink when loading data to primary key table. If false, will treat the record as a delete operation to StarRocks table.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:99]: image
See the comment on Line 90.
Also note to change "primary key table" to "Primary Key tables".

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:99]: image
This sentence does not make sense. Change to -> If this parameter is set to `false`, xxx.
And I can't understand what "treat the record as a delete operation to StarRocks table" means.

File: docs/loading/Flink-connector-starrocks.md [hunk 15]
```diff
@@ -1,173 +1,647 @@
...
+| sink.properties.*                 | No           | NONE              | The parameters that are used to control Stream Load behavior.  For example, the parameter `sink.properties.format` specifies the format used for stream load, such as CSV or JSON. For a list of supported parameters and their descriptions, see [STREAM LOAD](https://docs.starrocks.io/en-us/latest/sql-reference/sql-statements/data-manipulation/STREAM%20LOAD).                                                                                                                                                                                                                                                                                                                                 |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:100]: image
extra space.

File: docs/loading/Flink-connector-starrocks.md [hunk 16]
```diff
@@ -1,173 +1,647 @@
...
+| sink.properties.format            | No           | csv               | The format used for stream load. The connector will transform each batch of data to the format before sending them to StarRocks. Valid values: csv and json.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:101]: image
-> Stream Load

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:101]: image
-> The Flink connector

File: docs/loading/Flink-connector-starrocks.md [hunk 17]
```diff
@@ -1,173 +1,647 @@
...
+| sink.properties.max_filter_ratio  | No           | 0                 | The maximum error tolerance of the stream load. It's the maximum percentage of data records that can be filtered out due to inadequate data quality. Valid values: 0 to 1. Default value: 0. See [Stream Load](https://docs.starrocks.io/en-us/latest/sql-reference/sql-statements/data-manipulation/STREAM%20LOAD) for details.                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:104]: image
Search for all "stream load" and see if they are correctly written in the doc.

File: docs/loading/Flink-connector-starrocks.md [hunk 18]
```diff
@@ -1,173 +1,647 @@
...
+
+- If data loading pauses, you can try to increase the memory of the Flink task.
 
 - If the preceding code runs as expected and StarRocks can receive data, but the data loading fails, check whether your machine can access the HTTP port of the backends (BEs) in your StarRocks cluster. If you can successfully ping the HTTP port returned by the execution of the SHOW BACKENDS command in your StarRocks cluster, your machine can access the HTTP port of the BEs in your StarRocks cluster. For example, a machine has a public IP address and a private IP address, the HTTP ports of frontends (FEs) and BEs can be accessed through the public IP address of the FEs and BEs, the IP address that is bounded with your StarRocks cluster is the private IP address, and the value of `loadurl` for the Flink task is the HTTP port of the public IP address of the FEs. The FEs forwards the data loading task to the private IP address of the BEs. In this example, if the machine cannot ping the private IP address of the BEs, the data loading fails.
+
+- If you specify the value as exactly-once, `sink.buffer-flush.max-bytes`, `sink.buffer-flush.max-bytes`, and `sink.buffer-flush.interval-ms` are invalid.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:148]: image
Whose value?
Change the sentence to -> If you set `sink.semantic` to `exactly-once`

File: docs/loading/Flink-connector-starrocks.md [hunk 19]
```diff
@@ -1,173 +1,647 @@
...
+ENGINE=OLAP
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Set up Flink environment
+
+- Download Flink binary [Flink 1.15.2](https://archive.apache.org/dist/flink/flink-1.15.2/flink-1.15.2-bin-scala_2.12.tgz), and unzip it to directory `flink-1.15.2`.
+- Download [connector 1.2.7](https://repo1.maven.org/maven2/com/starrocks/flink-connector-starrocks/1.2.7_flink-1.15/flink-connector-starrocks-1.2.7_flink-1.15.jar), and put it into the directory `flink-1.15.2/lib`.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:178]: image
-> Flink connector

File: docs/loading/Flink-connector-starrocks.md [hunk 20]
```diff
@@ -1,173 +1,647 @@
...
+### Run with Flink SQL
+
+- Run the following command to start a Flink SQL client.
+
+    ```shell
+    ./bin/sql-client.sh
+    ```
+
+- Create a Flink table `score_board`, and insert values into the table via Flink SQL Client.
+Note you must define the primary key in the Flink DDL if load to a StarRocks primary key table. It's optional for other types of StarRocks table.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:195]: image
-> if you want to load data into a Primary Key table of StarRocks

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:195]: image
-> tables

File: docs/loading/Flink-connector-starrocks.md [hunk 21]
```diff
@@ -1,173 +1,647 @@
...
+        'username' = 'root',
+        'password' = ''
+    );
+
+    INSERT INTO `score_board` VALUES (1, 'starrocks', 100), (2, 'flink', 100);
+    ```
+
+### Run with Flink DataStream
+
+There are several ways to implement a Flink DataStream job according to the type of the input records, such as a csv Java `String`, a json Java `String` or a custom Java object.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:218]: image
CSV, JSON

File: docs/loading/Flink-connector-starrocks.md [hunk 22]
```diff
@@ -1,173 +1,647 @@
...
+            "1\tstarrocks-csv\t100",
+            "2\tflink-csv\t100"
+    };
+    DataStream<String> source = env.fromElements(records);
+
+    // Configure the connector with the required properties, and you also need to add properties
+    // "sink.properties.format" and "sink.properties.column_separator" to tell the connector the
+    // input records are csv-format, and the separator is "\t". You can also use other separators
+    // in the records, but remember to modify the "sink.properties.column_separator" correspondingly
+    StarRocksSinkOptions options = StarRocksSinkOptions.builder()
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:235]: image
there are many (not just the ones in the above figure) comments in code snippets, and these comments seem to use bad wording. I suggest optimizing all of them at least make them readable.

File: docs/loading/Flink-connector-starrocks.md [hunk 23]
```diff
@@ -1,173 +1,647 @@
...
+            internalRow[1] = rowData.name;
+            internalRow[2] = rowData.score;
+            // Only need for StarRocks primary key table. Set the last
+            // element to tell whether the record is a UPSERT or DELETE.
+            internalRow[internalRow.length - 1] = StarRocksSinkOP.UPSERT.ordinal();
+        }
+    }  
+    ```
+
+## Best Practices
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:358]: image
Best practices

File: docs/loading/Flink-connector-starrocks.md [hunk 24]
```diff
@@ -1,173 +1,647 @@
...
+
+### Load data to primary key table
+
+This section will show how to load data to StarRocks primary key table to achieve partial update, and conditional update.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:362]: image
see previous comments

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:362]: image
partial updates and conditional updates

File: docs/loading/Flink-connector-starrocks.md [hunk 25]
```diff
@@ -1,173 +1,647 @@
...
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Partial update
+
+This example will show how to load data only to columns `id` and `name`.
+
+1. Insert initial data to StarRocks table in MySQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:389]: image
-> Insert to the StarRocks table `score_board` in your MySQL client.

File: docs/loading/Flink-connector-starrocks.md [hunk 26]
```diff
@@ -1,173 +1,647 @@
...
+    +------+-----------+-------+
+    | id   | name      | score |
+    +------+-----------+-------+
+    |    1 | starrocks |   100 |
+    |    2 | flink     |   100 |
+    +------+-----------+-------+
+    2 rows in set (0.02 sec)
+    ```
+
+2. Create a Flink table `score_board` in Flink SQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:404]: Add a period (.) at the end of the sentence.

File: docs/loading/Flink-connector-starrocks.md [hunk 27]
```diff
@@ -1,173 +1,647 @@
...
+
+   - Define the DDL which only includes columns `id` and `name`
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:406]: Add a period (.) at the end of the sentence.

File: docs/loading/Flink-connector-starrocks.md [hunk 28]
```diff
@@ -1,173 +1,647 @@
...
+   - Set the option `sink.properties.partial_update` to `true` which tells the connector to do partial update.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:407]: image
-> Flink connector. See to all "connectors" in this doc

File: docs/loading/Flink-connector-starrocks.md [hunk 29]
```diff
@@ -1,173 +1,647 @@
...
+   - If the connector version <= 1.2.7, also need to set the option `sink.properties.columns` to `id,name,__op` which tells the connector the columns to update. Note you need append `__op` at the end, and this field is used to specify UPSERT/DELETE operation, but its value is set by the connector automatically.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:408]: image
-> is earlier than or equal to

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:408]: image
"who" need to???

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:408]: image
-> need to append?

Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:408]: image
-> specify an UPSERT or DELETE operation

File: docs/loading/Flink-connector-starrocks.md [hunk 30]
```diff
@@ -1,173 +1,647 @@
...
+        'table-name' = 'score_board',
+        'username' = 'root',
+        'password' = '',
+        'sink.properties.partial_update' = 'true',
+        -- only need for connector version <= 1.2.7
+        'sink.properties.columns' = 'id,name,__op'
+    ); 
+    ```
+
+3. Insert data to the table in Flink SQL client, and only update the column `name`
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:429]: image
period

File: docs/loading/Flink-connector-starrocks.md [hunk 31]
```diff
@@ -1,173 +1,647 @@
...
+
+    ```SQL
+    INSERT INTO `score_board` VALUES (1, 'starrocks-update'), (2, 'flink-update');
+    ```
+
+4. Query the StarRocks table in mysql client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:435]: period

File: docs/loading/Flink-connector-starrocks.md [hunk 32]
```diff
@@ -1,173 +1,647 @@
...
+    +------+-----------+-------+
+    | id   | name      | score |
+    +------+-----------+-------+
+    |    1 | starrocks |   100 |
+    |    2 | flink     |   100 |
+    +------+-----------+-------+
+    2 rows in set (0.02 sec)
+    ```
+
+2. Create a Flink table `score_board` in the following ways
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:469]: image
periods

File: docs/loading/Flink-connector-starrocks.md [hunk 33]
```diff
@@ -1,173 +1,647 @@
...
+  
+    - Define the DDL including all of columns
+    - Set the option `sink.properties.merge_condition` to `score` which tells the connector to use the column `score`
+    as the condition
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:473]: image
periods

File: docs/loading/Flink-connector-starrocks.md [hunk 34]
```diff
@@ -1,173 +1,647 @@
...
+        'load-url' = '127.0.0.1:8030',
+        'database-name' = 'test',
+        'table-name' = 'score_board',
+        'username' = 'root',
+        'password' = '',
+        'sink.properties.merge_condition' = 'score'
+    );
+    ```
+
+3. Insert data to the table in Flink SQL client, and update id 1 with a smaller score, and id 2 with a larger score.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:493]: image
the sentence does not make sense.

File: docs/loading/Flink-connector-starrocks.md [hunk 35]
```diff
@@ -1,173 +1,647 @@
...
+    +------+-----------+-------+
+    | id   | name      | score |
+    +------+-----------+-------+
+    |    1 | starrocks |    99 |
+    |    2 | flink     |   101 |
+    +------+-----------+-------+
+    2 rows in set (0.02 sec)
+    ```
+
+  You can see that only the score for id 2 changes, and the score for id 1 does not change.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:512]: image
does not make sense

File: docs/loading/Flink-connector-starrocks.md [hunk 36]
```diff
@@ -1,173 +1,647 @@
...
+    CREATE TABLE `test`.`page_uv` (
+      `page_id` INT NOT NULL COMMENT 'page ID',
+      `visit_date` datetime NOT NULL COMMENT 'access time',
+      `visit_users` BITMAP BITMAP_UNION NOT NULL COMMENT 'user ID'
+    ) ENGINE=OLAP
+    AGGREGATE KEY(`page_id`, `visit_date`)
+    DISTRIBUTED BY HASH(`page_id`);
+    ```
+
+2. Create a Flink table in Flink SQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:533]: image
periods

File: docs/loading/Flink-connector-starrocks.md [hunk 37]
```diff
@@ -1,173 +1,647 @@
...
+
+    `visit_user_id` is `BIGINT` in Flink, and we want to load it to the column `visit_users` of StarRocks table. Note that for the Flink DDL.
+    - define the `visit_user_id` instead of `visit_users` because Flink does not support `BITMAP`
+    - set the option `sink.properties.columns` to `page_id,visit_date,user_id,visit_users=to_bitmap(visit_user_id)` which tells the connector the column mapping
+    between Flink table and StarRocks table. It uses the [`to_bitmap`](https://docs.starrocks.io/en-us/latest/sql-reference/sql-functions/bitmap-functions/to_bitmap)
+   function to convert the data of `BIGINT` type into `BITMAP` type.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:539]: image
I really think these sentences need futher improvements.

File: docs/loading/Flink-connector-starrocks.md [hunk 38]
```diff
@@ -1,173 +1,647 @@
...
+        'load-url' = '127.0.0.1:8030',
+        'database-name' = 'test',
+        'table-name' = 'page_uv',
+        'username' = 'root',
+        'password' = '',
+        'sink.properties.columns' = 'page_id,visit_date,visit_user_id,visit_users=to_bitmap(visit_user_id)'
+    );
+    ```
+
+3. Load data into Flink table in Flink SQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:558]: image
period

File: docs/loading/Flink-connector-starrocks.md [hunk 39]
```diff
@@ -1,173 +1,647 @@
...
+    CREATE TABLE `hll_uv` (
+      `page_id` INT NOT NULL COMMENT 'page ID',
+      `visit_date` datetime NOT NULL COMMENT 'access time',
+      `visit_users` HLL HLL_UNION NOT NULL COMMENT 'user ID'
+    ) ENGINE=OLAP
+    AGGREGATE KEY(`page_id`, `visit_date`)
+    DISTRIBUTED BY HASH(`page_id`);
+    ```
+
+2. Create a Flink table in Flink SQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:602]: image
period

File: docs/loading/Flink-connector-starrocks.md [hunk 40]
```diff
@@ -1,173 +1,647 @@
...
+
+   `visit_user_id` is `BIGINT` in Flink, and we want to load it to the column `visit_users` of StarRocks table. Note that for the Flink DDL
+    - define the `visit_user_id` instead of `visit_users` because Flink does not support `BITMAP`
+    - set the option `sink.properties.columns` to `page_id,visit_date,user_id,visit_users=hll_hash(visit_user_id)` which tells the connector the column mapping
+      between Flink table and StarRocks table. It uses the [`hll_hash`](../sql-reference/sql-functions/aggregate-functions/hll_hash)
+      function to convert the data of `BIGINT` type into `HLL` type.
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:608]: image
At least capitalize the first letter and use a period at the sentence end.

File: docs/loading/Flink-connector-starrocks.md [hunk 41]
```diff
@@ -1,173 +1,647 @@
...
+        'load-url' = '127.0.0.1:8030',
+        'database-name' = 'test',
+        'table-name' = 'hll_uv',
+        'username' = 'root',
+        'password' = '',
+        'sink.properties.columns' = 'page_id,visit_date,visit_user_id,visit_users=hll_hash(visit_user_id)'
+    );
+    ```
+
+3. Load data into Flink table in Flink SQL client
```
Comment (amber-create) [inline docs/loading/Flink-connector-starrocks.md:627]: image
period

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by amber-create — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**